### PR TITLE
Implement exam quiz

### LINKED ISSUE #60: [Feature] Exam Mode Quiz UI
**Is your feature request related to a problem? Please describe.**
As a student preparing for the Azure Fundamentals certification, I value a user-friendly and engaging interface when I enter exam mode. Currently the exam mode only has the intro screen, the next one is the quiz itself. 

**Describe the solution you'd like**
To enhance the exam experience, we need to implement the UI for the exam mode that includes the following components:
 - **Question Display:** Each question should be presented clearly and legibly. The practice mode uses Next image, for exam mode let's keep it simple text that can't be selected.
 - **Answer Selection:** I should be able to select my answer choice by clicking the option bar. The selected choice should be visually indicated.
 - **Timer:** A timer displaying the remaining time in "mm:ss" format should be prominently visible throughout the exam in the top right corner. The timer should start automatically when the quiz appears and stop when the time limit is reached or the user has answered the final question.
- **Progress Indicator:** An indicator showing the progress, such as "X of 50," should help keep track of the progress. It should be positioned in the top left corner and it should update dynamically as the user move through the questions.
- **Next Question Button:** A clearly labeled "Next Question" button should be provided for the user to move to the next question and save the selected answer/s. The button should be disabled if no answer is selected.
- **Skip Question Button:** A clearly labeled "Skip Question" button should be provided for the user to move to the next question without selecting any answer. The skipped questions should be visible at the end of the exam.

**Additional context**
The practice mode already has the quiz component made, which it's a good starting point. 

image
You can find the link to figma file [here.](https://www.figma.com/file/ckNeex3dSR7Y32bPfJHiQQ/Azure-Fundamentals?type=design&node-id=417%3A10&mode=design&t=iEDVcKy710touMia-1)

### PR DESCRIPTION
**Pull request by masterHAWK99**
Fixes #60  

- [x] I have read and agree to the [Code of Conduct](https://github.com/eduardconstantin/azure-fundamentals/blob/main/CODE_OF_CONDUCT.md), and have followed the [Contributing guidelines](https://github.com/eduardconstantin/azure-fundamentals/blob/main/CONTRIBUTING.md).

## Pull Request Type

- [x] New feature

## Summary

I implemented a quiz system by simplifying an existing component. However, while creating a PR, I noticed that I had unintentionally implemented features from other issues.

### CODE REVIEW

**Review by eduardconstantin — CHANGES_REQUESTED** [review]
This looks really good, there are a few changes to be made

File: components/ExamResult.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                status === "success" ? "text-green-500" : "text-red-500"
+              }
+            >
+              {points}
+            </span>
+            /100
+          </p>
+        </div>
+      </div>
+      <p className="text-white text-lg">{render}</p>
```
Comment (eduardconstantin) [inline components/ExamResult.tsx:47]:
```suggestion
      <p className="text-white text-lg mb-12">{render}</p>
```

**Review by eduardconstantin — CHANGES_REQUESTED** [review]

File: pages/exam/index.tsx [hunk 2]
```diff
@@ -17,66 +20,176 @@ const questionsQuery = gql`
...
+          property="og:title"
+          content="Azure Fundamentals - Exam"
+          key="title"
+        />
       </Head>
       <div className="py-10 px-5 sm:p-10 w-5/6 sm:w-1/2 bg-slate-800 border-2 border-slate-700 rounded-lg">
         <div className="w-full flex flex-col xl:flex-row justify-between items-center">
-          <p className="text-white font-bold text-2xl">0/{numberOfQuestions}</p>
+          <p className="text-white font-bold text-2xl">
+            {currentQuestionIndex}/{numberOfQuestions}
```
[the remark this answers is not in the document]
Comment (masterHAWK99) [reply pages/exam/index.tsx:110]: In practice mode, the index starts from 0. Would you like me to apply this there as well?

File: pages/exam/index.tsx [hunk 3]
```diff
@@ -17,66 +20,179 @@ const questionsQuery = gql`
...
+  useEffect(() => {
+    if (remainingTime === "00:00") {
+      checkPassed();
+    }
+  }, [checkPassed, remainingTime]);
+
   if (error) return <p>Oh no... {error.message}</p>;
 
-  const numberOfQuestions = data.randomQuestions.length || 0;
+  const numberOfQuestions = 30;
```
Comment (eduardconstantin) [inline pages/exam/index.tsx:67]: Don't hardcode the number of questions, use the length from the query

**Review by eduardconstantin — CHANGES_REQUESTED** [review]

File: pages/exam/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (masterHAWK99) [reply pages/exam/index.tsx:110]: Sure, but for both?

[the remark this answers is not in the document]
Comment (eduardconstantin) [reply pages/exam/index.tsx:110]: Yes

**Review by eduardconstantin — APPROVED** [review]
Thank you!

**Pull request merged**